vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/SocketConnectionBase.java

```diff
@@ -70,7 +72,8 @@ public SocketConnectionBase(NetSocketInternal socket,
     this.socket = socket;
     this.context = context;
     this.pipeliningLimit = pipeliningLimit;
-    this.psCache = cachePreparedStatements ? new PreparedStatementCache(this, preparedStatementCacheSize) : null;
+    this.paused = false;
+    this.psCache = cachePreparedStatements ? new PreparedStatementCache(preparedStatementCacheSize) : null;
     this.preparedStatementCacheSqlLimit = preparedStatementCacheSqlLimit;
```

Comment:
we should use this option in the cache to make it work like before

Reply:
I mean `preparedStatementCacheSqlLimit` here

Reply:
why is this feature useful ? can we just instead remove it ?

Reply:
I added a generic filter for this, see the latest commit